prometheus/handler.go

```diff
@@ -0,0 +1,136 @@
+package prometheus
+
+import (
+	"compress/gzip"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/segmentio/stats"
+)
+
+// Handler is a type that bridges the stats API to a prometheus-compatible HTTP
+// endpoint.
+//
+// Typically, a program creates one Handler, registers it to the stats package,
+// and adds it to the muxer used by the application under the /metrics path.
+type Handler struct {
+	// A map of metric name to histograms buckets used by the handler to record
+	// observed values.
+	//
+	// The buckets are of list of upper limits used to group the observed
+	// values.
+	//
+	// If a bucket is missing for a metric name the observed value is ignored.
+	HistgramBuckets map[string][]float64
```

Comment:
This is what I'm most unsure about, the way prometheus handles histograms is a bit different than datadog, basically you specify a set of buckets for your values and the library has to maintain counters for each buckets.

It seems like the datadog histograms are closer to the concept of summaries in prometheus (using percentiles), but the prometheus histograms also seem to allow more powerful support for aggregations.

Another point that I'm unsure about is on the design of having the mapping in the handler. I think it makes sense because it's a prometheus-specific concept but I also feel weird about having it decoupled from the place where metrics are being produced. I'm also worried about having to introduce potentially breaking changes if we need to expose the concept of histogram buckets in the top-level API.

Please share your thoughts, the last thing I want is making breaking changes, but making software that is intuitive to use is also very important.